storage/google_storage.py

```diff
@@ -74,7 +84,10 @@ def save_to_directory(self, directory_path: str) -> None:
 
             if not relative_path[-1] == "/":
                 unversioned_blob = bucket.blob(blob.name)
-                retry.attempt(unversioned_blob.download_to_filename, local_file_path)
+                try:
```

Comment:
This isn't what I had in mind for `save_to_directory`, but I think it's still a good thing to do.

I was thinking that we would raise `NotFoundError` if the `list_blobs` call on line 77 returns an iterator of 0 blobs. I think this would be more similar to how S3 behaves.

However, if `list_blobs` yields one or more blobs, but downloading one of the blobs fails because it doesn't exist anymore, we should raise a `NotFoundError` for that, also.